### PR TITLE
Sorting Algorithm Method

### PR DESCRIPTION
**Pull request by alvaropazz**
-This project consists of two files:
*bubble_sort: it sorts an array of numbers in an ascending order
*bubble_sort_by: it sorts an array of strings by their length in ascending order

### CODE REVIEW

**Comment by pelzolga123** [comment]
## Hi Team!  
### You did a great job so far :+1: 
Just a few things left to fix before this project can be approved:
* [ ] Bubble_sort_by should return an error, if no block is given. For example:
`bubble_sort_by(%w[hi hello hey])` should return an error `no block given (yield)`
* [ ] Files not related to the project should not be committed and added to .gitignore (e.g. `.DS_Store`, `.vscode`) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
* [ ] Please add brief instruction on how to set up and run your project into README.md file.

### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

File: bubble_sort_by.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+        swap = true
+      end
+    end
+    break unless swap
+  end
+  arr
+end
+
+puts bubble_sort_by(%w[hi footbal rugby hello hey])
```
Comment (pelzolga123) [inline bubble_sort_by.rb]: Please kindly remove the debugging code. Only the method definition should be present.

File: bubble_sort.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+def bubble_sort(arr)
+  (0..arr.length - 1).each do |a|
+    ((a + 1)..arr.length - 1).each do |b|
+      arr[a], arr[b] = arr[b], arr[a] if arr[a] > arr[b]
+    end
+  end
+  p arr
+end
+
+bubble_sort([99, 100, 22, 7, 2, 1, 4, 5, 3])
```
Comment (pelzolga123) [inline bubble_sort.rb]: Please kindly remove the debugging code. Only the method definition should be present.

**Comment by thsvr** [comment]
### **STATUS: PROJECT APPROVED** :1st_place_medal: :1st_place_medal: :1st_place_medal: 

Hi @alvaropazz and @ugochukwuomeje :smile_cat: 

Congrats for the hard work, you guys did all the requests of Olga(TSE) and your project was approved :fireworks: :fireworks: :fireworks:

**Suggestion for the next projects:** 

:v: The development branch has a special usage in GitFlow. It means that you guys have to set development branch as your default branch and then create a feature branch with a descriptive name. To understand it clearly, please read this article: [GitFlow](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow) :smile_cat: :smile_cat: 
### 
**Happy coding and good luck on the next project** :computer: :computer: 

_Project reviewed by Thaís Vieira (TSE)_

**Pull request merged**